zk-sdk/src/encryption/auth_encryption.rs

```diff
@@ -117,12 +123,14 @@ impl AeKey {
     }
 
     /// Derive an authenticated encryption key from a signature.
+    #[cfg(not(target_arch = "wasm32"))]
     pub fn new_from_signature(signature: &Signature) -> Result<Self, Box<dyn error::Error>> {
         let seed = Self::seed_from_signature(signature);
         Self::from_seed(&seed)
     }
 
     /// Derive a seed from a signature used to generate an authenticated encryption key.
+    #[cfg(not(target_arch = "wasm32"))]
```

Comment:
What's the reason to exclude some of these functions? I can imagine that `&dyn Signer` doesn't make sense, but this one seems reasonable

Reply:
The problem is that if I include `solana-sdk` as part of the wasm target build, then the resulting wasm binary seems to include types like `Pubkey` or `Signature` that are not specific to the `zk-sdk` (https://github.com/rustwasm/wasm-bindgen/issues/3759). So I manually added `#[cfg(not(target_arch = "wasm32"))]` to all the places that depend on `solana-sdk` to prevent unnecessary bloat.

It might make sense to export this function for wasm targets later on (and in that case, it seems that we would need to depend on `solana-sdk`), but at least for this PR, I excluded it.